src/main/kotlin/org/rust/ide/inspections/UnnecessaryParenthesisInspection.kt

```diff
@@ -0,0 +1,67 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFixBase
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustBinaryExpr
+import org.rust.lang.core.psi.RustIfExpr
+import org.rust.lang.core.psi.RustParenExpr
+import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustParenExprImpl
+import org.rust.lang.core.psi.util.parentRelativeRange
+
+class UnnecessaryParenthesisInspection : RustLocalInspectionTool() {
+    override fun getDisplayName(): String = "Unnecessary parenthesis"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustVisitor() {
+            override fun visitParenExpr(expr: RustParenExpr) {
+                val subExpr = expr.expr
+                when (subExpr) {
+                    !is RustBinaryExpr -> {
+                        if (expr.parent !is RustIfExpr)
+                            holder.registerProblem(expr, "Redundant parenthesis around unambiguous expression", RemoveParenthesis)
+                    }
+                }
+            }
+
+            override fun visitIfExpr(expr: RustIfExpr) {
+                for (e in expr.exprList) {
+                    // Why is this a... list of expressions? Under what circumstances can an if have more than one?
```

Comment:
This is because the `else` branch can be either a block, an `if` or an `if_else` expression. So the grammar uses the common ancestor `expr` for the else branch.